integration/presto/src/main/java/org/apache/carbondata/presto/readers/SliceStreamReader.java

```diff
@@ -142,5 +144,17 @@ public SliceStreamReader(int batchSize, DataType dataType,
 
   @Override public void reset() {
     builder = type.createBlockBuilder(null, batchSize);
+    this.isLocalDict = false;
+  }
+
+  @Override public void putInt(int rowId, int value) {
+    Object data = DataTypeUtil
```

Comment:
Direct overriding does not create problem to local dictionary? how do you handle local dictionary here?

Reply:
putInt() will not be called incase of local dictionary as setDictionary() itself is filling all the values array.  Hence no impact with change to local dictionary. 
Also local dictionary UT are present and running fine after the changes